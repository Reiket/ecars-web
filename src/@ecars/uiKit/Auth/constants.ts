import {withBlockClass} from 'ecars-web-lib';
import {AuthPageWrapper} from '@ecars/uiKit/Auth/components/AuthPageWrapper';

export const AuthPageWrapperHOC = withBlockClass(AuthPageWrapper, 'auth');