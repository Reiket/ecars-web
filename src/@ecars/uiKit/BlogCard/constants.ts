import type {CategoriesListType} from 'ecars-web-lib';

export const BLOG_CARD_DIRECTION = Object.freeze({
  ROW: 'row',
  COLUMN: 'column',
});

export type BlogCardDirectionType = (typeof BLOG_CARD_DIRECTION)[keyof typeof BLOG_CARD_DIRECTION];

export interface BlogCardContentProps {
  title: string;
  category: CategoriesListType;
  description?: string;
}

export interface BlogCardImageProps {
  imageUrl: string;
  alt?: string;
}

export const BLOG_CARD_TEST_ID = 'blogCardTestId';
