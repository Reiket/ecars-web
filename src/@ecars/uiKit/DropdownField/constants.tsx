import type {DropdownOption} from 'ecars-web-lib';

interface TopbarDropdownsConfig {
  label: string;
  options: DropdownOption[];
}

const shipToConfig: DropdownOption[] = [
  {
    value: 'ca',
    name: 'Canada',
    label: (
      <img
        src="./assets/images/flags/canada.svg"
        alt="ca"
      />
    ),
  },
  {
    value: 'us',
    name: 'USA',
    label: (
      <img
        src="./assets/images/flags/usa.svg"
        alt="usa"
      />
    ),
  },
  {
    value: 'eu',
    name: 'Europe',
    label: (
      <img
        src="./assets/images/flags/europe.svg"
        alt="eu"
      />
    ),
  },
];

const currencyConfig: DropdownOption[] = [
  {
    value: 'usd',
    name: 'US Dollar',
  },
  {
    value: 'eur',
    name: 'Euro',
  },
  {
    value: 'gbp',
    name: 'British Pound',
  },
];

export const topbarDropdownsConfig: TopbarDropdownsConfig[] = [
  {
    label: 'Ship to:',
    options: shipToConfig,
  },
  {
    label: 'Currency:',
    options: currencyConfig,
  },
];
