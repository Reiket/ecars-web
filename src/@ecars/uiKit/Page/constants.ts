export const PAGE_CONTAINER_SIZE = Object.freeze({
  SM: 'sm',
  MD: 'md',
  LG: 'lg',
});

export type PageContainerSize = (typeof PAGE_CONTAINER_SIZE)[keyof typeof PAGE_CONTAINER_SIZE];
