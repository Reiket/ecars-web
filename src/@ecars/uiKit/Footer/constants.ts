import {withBlockClass} from 'ecars-web-lib';
import {FooterWrapper} from '@ecars/uiKit/Footer/components/FooterWrapper';
import {socialLinks} from '@ecars/uiKit/SocialList/constants';
import {PageUrls} from '@ecars/constants/page-urls';
import {FooterCopyWrapper} from '@ecars/uiKit/Footer/components/FooterCopyWrapper';

export const FooterWrapperHOC = withBlockClass(FooterWrapper, 'footer');
export const FooterCopyWrapperHOC = withBlockClass(FooterCopyWrapper, 'copy');
export const footerSocialLinks = socialLinks.filter((item) => !('label' in item));
export const footerContactsLinks = socialLinks.filter((item) => 'label' in item);

export const copyLinks = [
  {
    label: 'Privacy Policy',
    href: PageUrls.PRIVACY,
  },
  {
    label: 'Terms & Conditions',
    href: PageUrls.TERMS,
  },
];

export interface FooterLink {
  label: string;
  to: string;
}

export const companyLinks: FooterLink[] = [
  {label: 'About Us', to: PageUrls.ABOUT},
  {label: 'Blog', to: PageUrls.Blog},
  {label: 'FAQ', to: PageUrls.FAQ},
];

export const carsLinks: FooterLink[] = [
  {label: 'Special offers', to: '/'},
  {label: 'New cars', to: PageUrls.NEWCARS},
  {label: 'Used cars', to: PageUrls.USEDCARS},
  {label: 'Brands', to: '/'},
];
