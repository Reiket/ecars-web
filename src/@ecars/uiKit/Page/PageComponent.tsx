import type {FC, ReactNode} from 'react';
import {Page} from '@ecars/uiKit/Page';
import type {PageContainerSize} from '@ecars/uiKit/Page/constants';
import type {ElementProps} from 'ecars-web-lib';

interface Props extends ElementProps {
  children: ReactNode;
  containerSize?: PageContainerSize;
}

export const PageComponent: FC<Props> = ({block, children, containerSize}) => (
  <Page.Section block={block}>
    <Page.Container
      block={block}
      size={containerSize}
    >
      {children}
    </Page.Container>
  </Page.Section>
);
