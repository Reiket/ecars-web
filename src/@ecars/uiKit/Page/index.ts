import {PageComponent} from '@ecars/uiKit/Page/PageComponent';
import {PageContainer} from '@ecars/uiKit/Page/components/PageContainer';
import {PageSection} from '@ecars/uiKit/Page/components/PageSection';

export const Page = Object.assign(PageComponent, {
  Container: PageContainer,
  Section: PageSection,
});
