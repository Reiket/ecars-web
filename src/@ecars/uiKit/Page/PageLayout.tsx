import type {FC} from 'react';
import {Fragment} from 'react';
import {Topbar} from '@ecars/uiKit/Topbar';
import {Header} from '@ecars/uiKit/Header';
import {Footer} from '@ecars/uiKit/Footer';
import {Outlet} from 'react-router';

export const PageLayout: FC = () => (
  <Fragment>
    <Topbar />
    <Header />
    <main>
      <Outlet />
    </main>
    <Footer />
  </Fragment>
);
