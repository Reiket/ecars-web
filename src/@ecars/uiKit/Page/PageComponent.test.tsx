import {PAGE_CONTAINER_SIZE} from '@ecars/uiKit/Page/constants';
import {Page} from '@ecars/uiKit/Page/index';
import {render} from '@testing-library/react';

describe('Page Component', () => {
  const containerSizes = Object.values(PAGE_CONTAINER_SIZE);
  containerSizes.forEach((size) => {
    test('render component', () => {
      const {container} = render(
        <Page
          containerSize={size}
          block="test"
        >
          1
        </Page>,
      );
      expect(container).toMatchSnapshot();
      const containerElement = container.querySelector('.container');
      expect(container).toBeInTheDocument();
      expect(containerElement).toHaveClass(`container--${size}`);
    });
  });
});
