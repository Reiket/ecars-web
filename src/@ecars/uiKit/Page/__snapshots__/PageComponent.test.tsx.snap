// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page Component > render component 1`] = `
<div>
  <section
    class="test__section section"
  >
    <div
      class="test__container container container--sm"
    >
      1
    </div>
  </section>
</div>
`;

exports[`Page Component > render component 2`] = `
<div>
  <section
    class="test__section section"
  >
    <div
      class="test__container container container--md"
    >
      1
    </div>
  </section>
</div>
`;

exports[`Page Component > render component 3`] = `
<div>
  <section
    class="test__section section"
  >
    <div
      class="test__container container container--lg"
    >
      1
    </div>
  </section>
</div>
`;
